Automattic/wp-calypso
# state-utils: Create package by extracting and TypeScriptify-ing create-selector

#### Changes proposed in this Pull Request

* Extract `create-selector` into a new `state-utils` package and update its tests to jest instead of chai
* Take a shot at typing it in TypeScript (may have really missed on this one but not sure)

#### Testing instructions



* Ensure unit tests pass
* Ensure build passes
* Smoke test the app to make sure it looks good

Note: I only make a miniscule change to the `createSelector` function's logic to satisfy TypeScript so this should be safe as long as the build is working. In any case, the unit tests are pretty comprehensive and are passing so 🤞🏼
